Code/Tools/AssetProcessor/native/AssetManager/AssetRequestHandler.cpp

```diff
@@ -107,6 +107,46 @@ namespace
                     auto resultMove = relocationInterface->Move(
                         messageData.m_message->m_fromPath, messageData.m_message->m_toPath, false, true, true, true);
 
+                    if (resultMove.IsSuccess())
+                    {
+                        AssetProcessor::RelocationSuccess success = resultMove.TakeValue();
+
+                        // The report can be too long for the AZ_Printf buffer, so split it into individual lines
+                        AZStd::string report =
+                            relocationInterface->BuildChangeReport(success.m_relocationContainer, success.m_updateTasks);
+                        AzFramework::StringFunc::Tokenize(report.c_str(), lines, "\n");
```

Comment:
I recommend looking at using AZ::StringFunc::TokenizeVisitor instead to split a string without performing any memory allocations 
See [XConsole::AddLineAppendWithPrevLine](https://github.com/o3de/o3de/blob/development/Code/Legacy/CrySystem/XConsole.cpp#L2459-L2477) for use a lambda visitor function to output each line

Reply:
Done.